content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst

```diff
@@ -2,78 +2,471 @@
+`TD04` for credit notes, with the **tax-exemption code** ranging from `N6.1` to `N6.9`.
+
+The *buyer* is required to issue themselves an invoice (i.e., **self-billing**) for the VAT due with
+:guilabel:`Tipo Documento` code `TD16`, which is sent through the :abbr:`SdI (Systema di
+Interscambio)` with the buyer *themselves* as **recipient** (using their own :guilabel:`Codice
+Destinatario`). That way, they can deduct their VAT. The **self-billed** invoice must be issued and
+recorded in the VAT sales register within **15 days** from receipt.
+
+.. _italy/grids:
+
+'Reverse Charge' tax grids
+--------------------------
+
+The Italian localization has a specific **tax grid** section for **reverse charge** taxes. These
+tax grids are identifiable by the :ref:`VJ <italy/grids>` tag, and can be found under
+:menuselection:`Accounting --> Reporting --> Audit Reports: Tax Report`.
+
+.. image:: italy/italy-grids.png
    :align: center
-   :alt: Electronic invoicing status (waiting for confirmation)
+   :alt: Italian reverse charge tax grids
+
+San Marino
+==========
+
+Invoices
+--------
+
+San Marino and Italy have special agreements on e-Invoicing operations. As such, **invoices** follow
+the regular **reverse charge** rules. Additional requirements are not enforced by Odoo, however, the
+user is requested by the **State** to:
+
+- Select a tax with the option :guilabel:`Has exoneration of tax (Italy)` ticked, and the
+  :guilabel:`Exoneration` set to `N3.3`;
+- Use the generic :abbr:`SdI (Sistema di Interscambio)` :guilabel:`Codice Destinatario` `2R4GT08`.
+  The invoice is then routed by a dedicated office in San Marino to the correct business.
+
+Bills
+-----
+
+When a **paper bill** is received from San Marino, any Italian company **must** submit that invoice
+to the **Agenzia delle Entrate** by indicating the e-Invoice's :guilabel:`Tipo Documento` field with
+the special value `TD28`.
+
+.. tabs::
+
+   .. tab:: `TD28`
+
+      Odoo exports a move as `TD28` if the following conditions are met:
+
+      - Is a vendor bill;
+      - At least one tax on the invoice lines targets the tax grids :ref:`VJ <italy/grids>`;
+      - The **country** of the partner is **San Marino**.
+
+Pubblica amministrazione (B2G)
+==============================
+
+.. warning::
+   Odoo does **not** support e-Invoicing to **any** partner belonging to the **Pubblica
+   amministrazione dell'Italia**. All the information below is just a reference for *future*
+   implementations.
+
+Digital qualified signature
+---------------------------
+
+For invoices and bills intended to the **Pubblica Amministrazione (B2G)**, a **Digital Qualified
+Signature** is required for all files sent through the :abbr:`SdI (Sistema di Interscambio)`. The
+**XML** file must be certified using a certificate that is either:
+
+- a **smart card**;
+- a **USB token**;
+- a **Hardware Security Module (HSM)**.
```

Comment:
This section could be about the external program above: 
external program (with smart card or usb token)